search/includes/classes/class-search.php

```diff
@@ -1283,6 +1283,20 @@ public function filter__jetpack_active_modules( $modules ) {
 		} ) ); // phpcs:ignore WordPress.WhiteSpace.DisallowInlineTabs.NonIndentTabsUsed
 	}
 
+	/**
+	 * Remove the marketing admin page for Jetpack Search if VIP Search is enabled
+	 */
+	function remove_jetpack_menu_search(): void {
+		if ( defined( 'VIP_ENABLE_VIP_SEARCH' ) && true === VIP_ENABLE_VIP_SEARCH ) {
```

Comment:
I think this check here is redundant because the action is only getting added if the query integration is enabled, meaning Search is 100% enabled.